zerver/lib/actions.py

```diff
@@ -2273,27 +2274,33 @@ def notify_subscriptions_removed(user_profile, streams, no_log=False):
 def bulk_remove_subscriptions(users, streams, acting_user=None):
     # type: (Iterable[UserProfile], Iterable[Stream], Optional[UserProfile]) -> Tuple[List[Tuple[UserProfile, Stream]], List[Tuple[UserProfile, Stream]]]
 
-    recipients_map = bulk_get_recipients(Recipient.STREAM,
-                                         [stream.id for stream in streams])  # type: Mapping[int, Recipient]
-    stream_map = {}  # type: Dict[int, Stream]
-    for stream in streams:
-        stream_map[recipients_map[stream.id].id] = stream
+    users = list(users)
+    streams = list(streams)
 
-    subs_by_user = dict((user_profile.id, []) for user_profile in users)  # type: Dict[int, List[Subscription]]
-    for sub in Subscription.objects.select_related("user_profile").filter(user_profile__in=users,
-                                                                          recipient__in=list(recipients_map.values()),
-                                                                          active=True):
-        subs_by_user[sub.user_profile_id].append(sub)
+    stream_ids = {stream.id for stream in streams}
+    stream_dict = {stream.id: stream for stream in streams}
+
+    info = get_bulk_stream_subscriber_info(users, streams)
```

Comment:
I kinda want a more descriptive variable name than `info` here.  Maybe `subscriber_info` is good enough?